sound/soc/sof/intel/hda.c

```diff
@@ -1117,7 +1117,7 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 
 			/* find out how many identical parts are expected */
 			for (k = 0; k < link->num_adr; k++) {
-				u64 adr2 = link->adr_d[i].adr;
+				u64 adr2 = link->adr_d[k].adr;
```

Comment:
Nice catch. The reason how it worked is that we only support the same codecs on a single link. i.e. adr_d[0] == adr_d[1] == adr_d[2] etc for mfg_id, part_id and link_id. So we can still count `expected_part_count` using the same adr_d[i].adr.

Reply:
@plbossart Maybe above should be added to the commit message.